openhands/server/session/agent_session.py

```diff
@@ -200,15 +203,28 @@ async def _create_runtime(
             if github_token
             else None
         )
-        self.runtime = runtime_cls(
-            config=config,
-            event_stream=self.event_stream,
-            sid=self.sid,
-            plugins=agent.sandbox_plugins,
-            status_callback=self._status_callback,
-            headless_mode=False,
-            env_vars=env_vars,
-        )
+
+        if runtime_cls == RemoteRuntime:
+            self.runtime = runtime_cls(
+                config=config,
+                event_stream=self.event_stream,
+                sid=self.sid,
+                plugins=agent.sandbox_plugins,
+                status_callback=self._status_callback,
+                headless_mode=False,
+                env_vars=env_vars,
+                user_id=self.user_id,  # Pass user id
```

Comment:
nit: maybe we could just do:
```
kwargs = {}
if runtime_cls == RemoteRuntime:
    kwargs['user_id'] = self.user_id
```
And then pass `**kwargs` in `runtime_cls` constructor to reduce some code duplication